server/sonar-server/src/main/java/org/sonar/server/rule/ws/SearchAction.java

```diff
@@ -539,4 +538,237 @@ public SearchResult setFacets(Facets facets) {
     }
   }
 
+  private static class SearchRequest {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this class so it has no more than 20 fields, rather than the 23 it currently has. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1820)